### PR TITLE
APM-901 | Respect Open PR

### PR DESCRIPTION
**Pull request by Sourceless**
This change adds a flag `--respect_prs` to `env_cleaner.py`, which checks the public GitHub API for any open pull requests that match the name of a proxy deployed on apigee. Any proxies without pull requests are then acted upon, and vice versa.

Unit tests are included.

### CODE REVIEW

**Comment by Sourceless** [comment]
Changing to draft because the unit test does not actually do anything.

**Comment by Sourceless** [comment]
Will add unit test to bug ticket, but this is needed now

File: env_cleaner.py [hunk 1]
```diff
@@ -33,6 +34,35 @@
 SANDBOX_MATCHER = re.compile(".+-sandbox$")
 SANDBOX_ANTIMATCHER = re.compile(".+-internal-qa-sandbox")
 
+# Map of repository names to service names
+REPO_NAMES = {
+    "personal-demographics-service-api": "personal-demographics",
```
Comment (ri-ch) [inline env_cleaner.py:39]: Presumably we would need to add the EPS repo and base path into here?

Comment (Sourceless) [reply env_cleaner.py:39]: Correct

**Pull request merged**